interfaces/core.go

```diff
@@ -265,20 +265,57 @@ type StaticInfo struct {
 	AppArmorUnconfinedSlots bool
 }
 
+type PlugServicesSnippetSection string
+
+const (
+	PlugServicesSnippetUnitSection    PlugServicesSnippetSection = "Unit"
+	PlugServicesSnippetServiceSection PlugServicesSnippetSection = "Service"
+)
+
+// PlugServiceSnippet describes a systemd service snippet to be generated
+// for a snap with a plug whose interface implements ServicePermanentPlug.
+type PlugServicesSnippet interface {
+	// Section is the target unit file section for the snippet to be
+	// injected in (i.e. [Unit], [Service]).
+	Section() PlugServicesSnippetSection
```

Comment:
Given that Section is pretty generic and String is very common, could we name this SystemdConfSection or something _longer_ than just Section alone?